packages/features/ee/round-robin/roundRobinReassignment.ts

```diff
@@ -0,0 +1,371 @@
+    }
+  })();
+
+  // If changed owner, also change destination calendar
+  const previousHostDestinationCalendar = hasOrganizerChanged
+    ? await prisma.destinationCalendar.findFirst({
+        where: {
+          userId: originalOrganizer.id,
+        },
+      })
+    : null;
+
+  const evt: CalendarEvent = {
+    organizer: {
+      name: organizer.name || "",
+      email: organizer.email,
+      language: {
+        locale: organizer.locale || "en",
+        translate: organizerT,
+      },
+      timeZone: organizer.timeZone,
+      timeFormat: getTimeFormatStringFromUserTimeFormat(organizer.timeFormat),
+    },
+    startTime: dayjs(booking.startTime).utc().format(),
+    endTime: dayjs(booking.endTime).utc().format(),
+    type: eventType.slug,
+    title: booking.title,
+    description: eventType.description,
+    attendees: attendeeList,
+    uid: booking.uid,
+    destinationCalendar,
+    team: {
+      members: teamMembers,
+      name: eventType.team?.name || "",
+      id: eventType.team?.id || 0,
+    },
+    customInputs: isPrismaObjOrUndefined(booking.customInputs),
+    ...getCalEventResponses({
+      bookingFields: eventType?.bookingFields ?? null,
+      booking,
+    }),
+  };
+
+  const credentials = await prisma.credential.findMany({
+    where: {
+      userId: organizer.id,
+    },
+    include: {
+      user: {
+        select: {
+          email: true,
+        },
+      },
+    },
+  });
+  const eventManager = new EventManager({ ...organizer, credentials: [...credentials] });
+
+  await eventManager.reschedule(
+    evt,
+    booking.uid,
+    undefined,
+    hasOrganizerChanged,
+    previousHostDestinationCalendar ? [previousHostDestinationCalendar] : []
+  );
+
+  // Send to new RR host
+  await sendRoundRobinScheduledEmails(evt, [
+    {
+      ...reassignedRRHost,
+      name: reassignedRRHost.name || "",
+      username: reassignedRRHost.username || "",
+      timeFormat: getTimeFormatStringFromUserTimeFormat(reassignedRRHost.timeFormat),
+      language: { translate: reassignedRRHostT, locale: reassignedRRHost.locale || "en" },
+    },
+  ]);
+  // Send to cancelled RR host
+  await sendRoundRobinCancelledEmails(evt, [
+    {
+      ...previousRRHost,
+      name: previousRRHost.name || "",
+      username: previousRRHost.username || "",
+      timeFormat: getTimeFormatStringFromUserTimeFormat(previousRRHost.timeFormat),
+      language: { translate: previousRRHostT, locale: previousRRHost.locale || "en" },
+    },
+  ]);
+
+  // Handle changing workflows with organizer
+  if (hasOrganizerChanged) {
+    const workflowReminders = await prisma.workflowReminder.findMany({
+      where: {
+        bookingUid: booking.uid,
+        method: WorkflowMethods.EMAIL,
+        workflowStep: {
+          action: WorkflowActions.EMAIL_HOST,
+          workflow: {
+            trigger: {
+              in: [
+                WorkflowTriggerEvents.BEFORE_EVENT,
+                WorkflowTriggerEvents.NEW_EVENT,
```

Comment:
NEW_EVENT triggers shouldn't have reminders saved as they are sent immediately, but I think it would make sense to also send out workflow emails to the new host for the NEW_EVENT trigger